### PR TITLE
Add initial tsc checks

### PR DESCRIPTION
**Pull request by jonathanKingston**
**Reviewer:** @kzar, @sammacbeth or @shakyShane

## Description:

Add initial tsc linting, depends on: https://github.com/duckduckgo/privacy-grade/pull/43

## Automated tests:
- [ ] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

### CODE REVIEW

File: shared/js/background/classes/site.es6.js [hunk 1]
```diff
@@ -105,14 +105,22 @@ class Site {
         return this.isProtectionEnabled() && this.enabledFeatures.includes(featureName)
     }
 
+    /**
+     * @param {import("@duckduckgo/privacy-grade/src/classes/trackers.js").TrackerData} t
```
Comment (jonathanKingston) [inline shared/js/background/classes/site.es6.js:109]: If the timing of approvals in privacy-grade is an issue we can change this to `@param {*}` for now.

File: shared/js/background/tab-manager.es6.js [hunk 2]
```diff
@@ -5,6 +5,7 @@ const browserWrapper = require('./wrapper.es6')
 
 class TabManager {
     constructor () {
+        /** @type {{ [key: number]: Tab }} */
```
Comment (shakyShane) [inline shared/js/background/tab-manager.es6.js:8]: NAB: prefer `Record<number, Tab>` here to indicate this type shouldn't inherit all of Object

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    "moduleResolution": "node",
+    "alwaysStrict": true,
+    "allowJs": true,
+    "checkJs": true,
+    "noEmit": true,
+    "allowSyntheticDefaultImports": true,
+    "resolveJsonModule": true,
+    "strict": true,
+    "noImplicitAny": false,
+    "strictNullChecks": true
```
Comment (shakyShane) [inline tsconfig.json:13]: nice!

**Review by shakyShane — APPROVED** [review]
LGTM

File: shared/js/background/tab-manager.es6.js (hunk 2 above)
Comment (jonathanKingston) [reply shared/js/background/tab-manager.es6.js]: Thanks, agreed that's even better. I avoided changing this to a Map which would provide similar benefits to this as I understand it.

File: shared/js/background/classes/site.es6.js (hunk 1 above)
Comment (jonathanKingston) [reply shared/js/background/classes/site.es6.js]: Made this change and will change it back once we land those changes.

**Pull request merged**